cluster_experiments/analysis_plan.py

```diff
@@ -0,0 +1,264 @@
+                            treatment_variant=treatment_variant,
+                            control_variant=control_variant,
+                            dimension_name=dimension.name,
+                            dimension_value=dimension_value,
+                        )
+
+                        ate = experiment_analysis.get_point_estimate(df=prepared_df)
+                        p_value = experiment_analysis.get_pvalue(df=prepared_df)
+                        std_error = experiment_analysis.get_standard_error(
+                            df=prepared_df
+                        )
+                        confidence_interval = (
+                            experiment_analysis.get_confidence_interval(
+                                df=prepared_df, alpha=self.alpha
+                            )
+                        )
+                        control_variant_mean = test.metric.get_mean(
+                            prepared_df.query(
+                                f"{self.variant_col}=='{control_variant.name}'"
+                            )
+                        )
+                        treatment_variant_mean = test.metric.get_mean(
+                            prepared_df.query(
+                                f"{self.variant_col}=='{treatment_variant.name}'"
+                            )
+                        )
+
+                        test_results.append(
+                            HypothesisTestResults(
+                                metric_alias=test.metric.alias,
+                                control_variant_name=control_variant.name,
+                                treatment_variant_name=treatment_variant.name,
+                                control_variant_mean=control_variant_mean,
+                                treatment_variant_mean=treatment_variant_mean,
+                                analysis_type=test.analysis_type,
+                                ate=ate,
+                                ate_ci_lower=confidence_interval.lower,
+                                ate_ci_upper=confidence_interval.upper,
+                                p_value=p_value,
+                                std_error=std_error,
+                                dimension_name=dimension.name,
+                                dimension_value=dimension_value,
+                                alpha=self.alpha,
+                            )
+                        )
+
+        return AnalysisPlanResults.from_results(test_results)
+
+    def prepare_data(
+        self,
+        data: pd.DataFrame,
+        variant_col: str,
+        treatment_variant: Variant,
+        control_variant: Variant,
+        dimension_name: str,
+        dimension_value: str,
+    ) -> pd.DataFrame:
+        """
+        Prepares the data for the experiment analysis pipeline
+        """
+        prepared_df = data.copy()
+
+        prepared_df = prepared_df.assign(total_dimension="total")
```

Comment:
should we have a dimension_col parameter?

Reply:
right now we have it, it is given by the name attribute of the Dimension object 👍